src/main/scala/com/typesafe/sbt/packager/docker/DockerPlugin.scala

```diff
@@ -497,12 +516,44 @@ object DockerPlugin extends AutoPlugin {
       override def buffer[T](f: => T): T = f
     }
 
-  def publishLocalDocker(context: File, buildCommand: Seq[String], log: Logger): Unit = {
+  def publishLocalDocker(context: File,
+                         buildCommand: Seq[String],
+                         execCommand: Seq[String],
+                         strategy: DockerPermissionStrategy,
+                         removeIntermediateImages: Boolean,
+                         log: Logger): Unit = {
     log.debug("Executing Native " + buildCommand.mkString(" "))
     log.debug("Working directory " + context.toString)
 
     val ret = sys.process.Process(buildCommand, context) ! publishLocalLogger(log)
 
+    // First let's see if there was a comment that tells us the id of the generated dockerfile
+    val headComments = IO.readLines(context / "Dockerfile").takeWhile(_.startsWith("# ")).map(_.substring(2))
```

Comment:
Why do we need a comment? Can't we just search for a `LABEL` definition that starts with `id` (or make it more unique, e.g. `snp-intermediate-id`?

After reading the docs I'm not sure why we need this at all. The filter combination seems restricted enough. The only issue I could think of is when 2 parallel builds are running and cleaning things up afterwards. Is this a real issue?

Reply:
> Why do we need a comment? Can't we just search for a `LABEL` definition that starts with `id` (or make it more unique, e.g. `snp-intermediate-id`?

I can change that (instead of a comment search for the `LABEL` definition and use `snp-intermediate-id`.

> After reading the docs I'm not sure why we need this at all. The filter combination seems restricted enough. The only issue I could think of is when 2 parallel builds are running and cleaning things up afterwards. Is this a real issue?

See my other comment. My point here is I do not want to introduce any sideffects _at all_ when packaging a docker image. I don't think it's a good idea that packaging could potentially clean images that have nothing to do with the **current** build process.